### PR TITLE
Add order list for logged user

### PR DESCRIPTION
**Pull request by zaiste**

### CODE REVIEW

**Comment by mmiszy** [comment]
Could you rebase @zaiste ?

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: src/app/(main)/orders/page.tsx [hunk 1]
```diff
@@ -60,7 +56,7 @@ export default function OrderPage() {
 											</div>
 											<div>
 												<dt className="font-medium text-slate-900">Total amount</dt>
-												<dd className="mt-1 font-medium text-slate-900">{0}</dd>
+												<dd className="mt-1 font-medium text-slate-900">{order.total?.gross.amount}</dd>
```
[the remark this answers is not in the document]
Comment (mmiszy) [reply src/app/(main)/orders/page.tsx:59]: Why `text-slate-900` instead of `text-neutral-900` ?

**Review by mmiszy — CHANGES_REQUESTED** [review]
Replace `slate` with `neutral`

**Comment by mmiszy** [comment]
This doesn't look too good:

Screenshot 2023-11-10 at 17 09 26

File: package.json [hunk 2]
```diff
@@ -16,18 +16,19 @@
...
 	"dependencies": {
 		"@adyen/adyen-web": "5.53.2",
-		"@adyen/api-library": "14.3.0",
+		"@adyen/api-library": "14.4.0",
 		"@headlessui/react": "1.7.17",
 		"@saleor/auth-sdk": "0.14.0",
 		"@stripe/react-stripe-js": "2.3.1",
 		"@stripe/stripe-js": "2.1.11",
 		"clsx": "2.0.0",
+		"date-fns": "2.30.0",
```
Comment (mmiszy) [inline package.json:25]: We don't need a library to format a date. We can just use `Intl`:
```
new Intl.DateTimeFormat('en-US').format(date)
```

File: src/app/(main)/orders/page.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import dynamic from "next/dynamic";
+
+export default function OrderPage() {
+	const OrderList = dynamic(() => import("../../../ui/components/OrderList"), { ssr: false });
```
Comment (mmiszy) [inline src/app/(main)/orders/page.tsx:4]: I think this should be outside the component body?
```suggestion
const OrderList = dynamic(() => import("../../../ui/components/OrderList").then((m) => m.OrderList), { ssr: false });

export default function OrderPage() {
```

File: src/ui/components/OrderList.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import { CurrentUserOrderListDocument, type CurrentUserOrderListQuery } from "@/gql/graphql";
+import { formatMoney } from "@/lib/graphql";
+
+// eslint-disable-next-line import/no-default-export
+export default function OrderList() {
+	const [{ data, fetching, error }] = useQuery<CurrentUserOrderListQuery>({
+		query: CurrentUserOrderListDocument.toString(),
+	});
+
+	console.log("here we go", fetching, data, error);
```
Comment (mmiszy) [inline src/ui/components/OrderList.tsx:14]:
```suggestion
```

File: src/ui/components/OrderList.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
+"use client";
+import { useQuery } from "urql";
+import Image from "next/image";
+import { format } from "date-fns";
+import { CurrentUserOrderListDocument, type CurrentUserOrderListQuery } from "@/gql/graphql";
+import { formatMoney } from "@/lib/graphql";
+
+// eslint-disable-next-line import/no-default-export
+export default function OrderList() {
```
Comment (mmiszy) [inline src/ui/components/OrderList.tsx:9]:
```suggestion
export function OrderList() {
```

File: src/ui/components/OrderList.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+		query: CurrentUserOrderListDocument.toString(),
+	});
+
+	console.log("here we go", fetching, data, error);
+
+	if (fetching) {
+		return <div>Loading...</div>;
+	}
+
+	if (!fetching) {
```
Comment (mmiszy) [inline src/ui/components/OrderList.tsx:20]: This condition is redundant because line 16 makes it impossible not to be true

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: src/app/(main)/orders/page.tsx (hunk 3 above)
Comment (zaiste) [reply src/app/(main)/orders/page.tsx]: @mmiszy named export doesn't work unfortunately

Comment (mmiszy) [reply src/app/(main)/orders/page.tsx]: It does work, we're already using in other places.

Comment (zaiste) [reply src/app/(main)/orders/page.tsx]: @mmiszy can you check on your side?

CleanShot 2023-11-11 at 14 00 10@2x

(with re-created `.next/`)

Comment (mmiszy) [reply src/app/(main)/orders/page.tsx]: Indeed there's a bug in Next.js :D Adding "use client" to the `page.tsx` fixes it.

Comment (zaiste) [reply src/app/(main)/orders/page.tsx]: :D 

Yeah, but I'd prefer not to set the entire page as `client`. Let's keep it as `default` for now with an eslint exception, there's an open bug report in the Next.js repo, let's track it and fix it once it's done upstream. WDYT?

**Review by mmiszy — APPROVED** [review]

File: src/app/(main)/orders/page.tsx (hunk 3 above)
Comment (mmiszy) [reply src/app/(main)/orders/page.tsx]: This whole page is client anyway because you're only rendering a single dynamic component.

**Pull request merged**